lua/lvim/core/telescope.lua

```diff
@@ -130,6 +130,12 @@ function M.setup()
     end)
   end
 
+  if lvim.builtin.notify.active then
+    pcall(function()
+      require("telescope").load_extension "notify"
+    end)
+  end
+
```

Comment:
It will popup a telescope window to see `notify` history and you can select one of them to see the detailed in another popup window.
![图片](https://user-images.githubusercontent.com/45989017/167446113-ecdca00b-a57f-4640-9200-890588b59064.png)
![图片](https://user-images.githubusercontent.com/45989017/167446140-abe24dee-9854-4e7e-887b-4350e40892d9.png)

Reply:
> umm, why not have this in the plugin's own setup function?

If put config to it's own setup function, we should ensure `telescope` is loaded before `notify` is loaded. Otherwise it will issue `E5108: Error executing lua .../start/telescope.nvim/lua/telescope/_extensions/init.lua:10: 'notify' extension doesn't exist`.

Reply:
> umm, why not have this in the plugin's own setup function?

We don't even need call `require("telescope").load_extension "notify"` because `notify` itself will do this if `telescope` is loaded. 
https://github.com/rcarriga/nvim-notify/blob/d4a01eedeb2e6c3d453f3aa0f1b303dd8611dc71/lua/notify/init.lua#L30-L33

So we only need lazy-load `notify` like this 
  {
    "rcarriga/nvim-notify",
    -- by after
    after = "telescope.nvim"
  },
or
  {
    "rcarriga/nvim-notify",
    -- by event
    event = "BufWinEnter",
  },

Reply:
I'm not sure we want to mess with lazy-loading for nvim-notify, it was pretty annoying to debug the errors the last time I dealt with it (it used to be lazy-loaded before https://github.com/LunarVim/LunarVim/pull/2359) 😢 

to me this works as expected

```diff
diff --git a/lua/lvim/core/notify.lua b/lua/lvim/core/notify.lua
index 2db4c4d5..fe37f5a0 100644
--- a/lua/lvim/core/notify.lua
+++ b/lua/lvim/core/notify.lua
@@ -61,6 +61,7 @@ function M.setup()
   local notify = require "notify"
 
   notify.setup(opts)
+  pcall(require("telescope").load_extension, "notify")
   vim.notify = notify
   Log:configure_notifications(notify)
 end
```

Reply:
> to me this works as expected

Cool! This works for me too. I think it is no bad to pre-config lvim builtins plugins which have telescope extenstion (`project.nvim`, `nvim-notify` and `telescope-fzf-native.nvim`) and put them in one place.

Reply:
oh I completely forgot we're doing the others here as well. I feel like it's always a gamble whether we need to call from the `config()` block of the plugin or here, but yeah let's give this a try!

for which-key, maybe `<leader>Lt`?
```diff
diff --git a/lua/lvim/core/which-key.lua b/lua/lvim/core/which-key.lua
index 3c3cc66..b8a1c9f 100644
--- a/lua/lvim/core/which-key.lua
+++ b/lua/lvim/core/which-key.lua
@@ -199,6 +199,7 @@ M.config = function()
           "<cmd>lua require('lvim.core.telescope.custom-finders').grep_lunarvim_files()<cr>",
           "Grep LunarVim files",
         },
+        t = { "<cmd>Telescope notify<cr>", "View Notifications" },
         k = { "<cmd>Telescope keymaps<cr>", "View LunarVim's keymappings" },
         i = {
           "<cmd>lua require('lvim.core.info').toggle_popup(vim.bo.filetype)<cr>",
```

Reply:
I personally use `<leader>sn`. Both `<leader>Lt` and  `<leader>Ln` is ok, maybe `n` is better to go with `notify`.

Reply:
sounds good, feel free to add the which-key integration then!

Reply:
Done!